sdks/go/pkg/beam/io/rtrackers/offsetrange/offsetrange_test.go

```diff
@@ -299,3 +301,299 @@ func TestTracker_TrySplit(t *testing.T) {
+		t.Errorf("wrong restriction tracked by tracker after TrySplit(0), got: %v, want: %v", tracker.GetRestriction().(Restriction), expPr)
+	}
+	expRes := Restriction{6, math.MaxInt64}
+	if r.(Restriction) != expRes {
+		t.Errorf("wrong residual TrySplit(0), got: %v, want: %v", r.(Restriction), expRes)
+	}
+
+	tracker, err = NewGrowableTracker(rest, &estimator)
+	if err != nil {
+		t.Fatalf("error creating new GrowableTracker: %v", err)
+	}
+	if !tracker.TryClaim(int64(5)) {
+		t.Fatal("tracker.TryClaim(int64(5)) should've claimed.")
+	}
+	estimator.SetEstimateRangeEnd(20)
+	p, r, _ = tracker.TrySplit(0)
+	if !tracker.IsDone() {
+		t.Fatal("tracker should have been done, but IsDone() returns false")
+	}
+	if p.(Restriction) != expPr {
+		t.Errorf("wrong primaries after TrySplit(0), got: %v, want: %v", p.(Restriction), expPr)
+	}
+	if r.(Restriction) != expRes {
+		t.Errorf("wrong residual TrySplit(0), got: %v, want: %v", r.(Restriction), expRes)
+	}
+}
+
+func TestGrowableTracker_TrySplit(t *testing.T) {
+	estimator := offsetRangeEndEstimator{EstimateRangeEnd: 0}
+	rest := Restriction{Start: 0, End: math.MaxInt64}
+	tracker, err := NewGrowableTracker(rest, &estimator)
+	if err != nil {
+		t.Fatalf("error in creating new GrowableTracker: %v", err)
+	}
+	if !tracker.TryClaim(int64(0)) {
+		t.Errorf("tracker.TryClaim(0) = %v, want: %v", false, true)
+	}
+
+	estimator.SetEstimateRangeEnd(16)
+
+	p, r, _ := tracker.TrySplit(0.5)
+	expPr := Restriction{0, 8}
+	if p.(Restriction) != expPr {
+		t.Errorf("wrong primaries after TrySplit(0.5), got: %v, want: %v", p.(Restriction), expPr)
+	}
+	if tracker.GetRestriction().(Restriction) != expPr {
+		t.Errorf("wrong restriction tracked by tracker after TrySplit(0.5), got: %v, want: %v", tracker.GetRestriction().(Restriction), expPr)
+	}
+	expRes := Restriction{8, math.MaxInt64}
+	if r.(Restriction) != expRes {
+		t.Errorf("wrong residual TrySplit(0.5), got: %v, want: %v", r.(Restriction), expRes)
+	}
+
+	estimator.SetEstimateRangeEnd(12)
+	p, r, _ = tracker.TrySplit(0.5)
+	expPr = Restriction{0, 4}
+	if p.(Restriction) != expPr {
+		t.Errorf("wrong primaries after TrySplit(0.5), got: %v, want: %v", p.(Restriction), expPr)
+	}
+	if tracker.GetRestriction().(Restriction) != expPr {
+		t.Errorf("wrong restriction tracked by tracker after TrySplit(0.5), got: %v, want: %v", tracker.GetRestriction().(Restriction), expPr)
+	}
+	expRes = Restriction{4, 8}
+	if r.(Restriction) != expRes {
+		t.Errorf("wrong residual TrySplit(0.5), got: %v, want: %v", r.(Restriction), expRes)
+	}
+
+	if tracker.TryClaim(int64(4)) {
+		t.Errorf("tracker.TryClaim(4) = %v, want: %v", true, false)
+	}
+}
+
+// TestGrowableTracker_Splits tests that TrySplit follows its contract, meaning that
+// splits don't lose any elements, split fractions are clamped to 0 or 1, and
+// that TrySplit always splits at the nearest integer greater than the given
+// fraction.
+func TestGrowableTracker_Splits(t *testing.T) {
```

Comment:
This is exactly what I'm thinking of. What's missing from these cases that is being covered by the tests above? We likely can migrate everything else into this test's table.